src/Illuminate/Database/Eloquent/Relations/Concerns/CanBeOneOfMany.php

```diff
@@ -191,11 +200,21 @@ protected function newOneOfManySubQuery($groupBy, $column = null, $aggregate = n
             $subQuery->groupBy($this->qualifyRelatedColumn($group));
         }
 
-        if (! is_null($column)) {
-            $subQuery->selectRaw($aggregate.'('.$subQuery->getQuery()->grammar->wrap($subQuery->qualifyColumn($column)).') as '.$subQuery->getQuery()->grammar->wrap($column.'_aggregate'));
+        if (! is_null($columns)) {
+            foreach ($columns as $key => $column) {
+                $aggregatedColumn = $subQuery->getQuery()->grammar->wrap($subQuery->qualifyColumn($column));
+
+                if ($key === 0) {
+                    $aggregatedColumn = "{$aggregate}({$aggregatedColumn})";
+                } else {
+                    $aggregatedColumn = "min({$aggregatedColumn})";
```

Comment:
Why `min` here? Can you explain the entire logic behind this PR and how it fixes the issue?

Reply:
The `min` purpose is to use an aggregate to select any value on a column which we already know all value are the same because it's the sub-level aggregate. We still need an aggregate because there's no `group by` clause on that column. The most correct solution would probably be to add a group by on that column, but I didn't manage to do it in the time I had.

The more global purpose of the PR is fully stated and detailed (and is demonstrated by the modification of the generated query) in the original PR : https://github.com/laravel/framework/pull/46309